src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,105 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_FACTOR = 2;
+
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        this(INITIAL_CAPACITY);
+    }
+
+    public MyHashMap(int capacity) {
+        this.table = new Node[capacity];
+        this.threshold = (int) (capacity * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= threshold) {
+            resize();
+        }
+        int index = getIndex(key);
+        Node<K,V> newNode = new Node<>(key, value);
 
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+        } else {
```

Comment:
```suggestion
        if (table[index] == null) {
            table[index] = newNode;
            size++;
            return;
        }
```
untested code will be easier to read, returning early is a preferred pattern